src/App.jsx

```diff
@@ -14,70 +15,70 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+const goodsObj = goods
+  .map((item, index) => ({ nameGoods: item, id: index + 1 }));
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export const App = () => {
+  const [checked, setChecked] = useState(9);
+  const [currentName, setCurrentName] = useState('Jam');
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
+  return (
+    <main className="section container">
+      {!checked
+        ? (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )
+        : (
+          <h1 className="title is-flex is-align-items-center">
+            {`${currentName} is selected`}
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              className="delete ml-3"
+              onClick={() => setChecked(null)}
+            />
+          </h1>
+        )
+      }
+      <table className="table">
+        <tbody>
+          {goodsObj.map((item) => {
+            const { nameGoods, id } = item;
+            const isChecked = checked === id;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+            const handlerClickButton = () => {
+              setChecked(isChecked ? null : id);
+              setCurrentName(nameGoods);
+            };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+            return (
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              <tr
+                key={id}
```

Comment:
![image](https://github.com/mate-academy/react_goods-selector-js/assets/96409563/0adc1cce-a838-42bd-bc9f-e442e98627d3)

Reply:
@polosanya 
I'm sorry, but I don't fully understand the approach I should use.

In the code, I use the id field from the goodsObj  as the key for the <tr>